rosidl_generator_java/resource/msg.cpp.em

```diff
@@ -35,44 +51,88 @@ array_list_jni_type = "java/util/ArrayList"
 cache = defaultdict(lambda: False)
 cache[msg_normalized_type] = msg_jni_type
 namespaced_types = set()
-includes = set()
+member_includes = set()
 for member in message.structure.members:
     type_ = member.type
     if isinstance(type_, AbstractNestedType):
         cache[list_normalized_type] = list_jni_type
         cache[array_list_normalized_type] = array_list_jni_type
         type_ = type_.value_type
         if isinstance(type_, BasicType):
-            includes.add('rosidl_generator_c/primitives_sequence.h')
-            includes.add('rosidl_generator_c/primitives_sequence_functions.h')
+            member_includes.add('rosidl_generator_c/primitives_sequence.h')
+            member_includes.add('rosidl_generator_c/primitives_sequence_functions.h')
 
     # We do not cache strings because java.lang.String behaves differently
     if not isinstance(type_, AbstractGenericString):
         cache[get_normalized_type(type_)] = get_jni_type(type_)
 
     if isinstance(type_, AbstractString):
-        includes.add('rosidl_generator_c/string.h')
-        includes.add('rosidl_generator_c/string_functions.h')
+        member_includes.add('rosidl_generator_c/string.h')
+        member_includes.add('rosidl_generator_c/string_functions.h')
 
     if isinstance(type_, AbstractWString):
-        includes.add('rosidl_generator_c/u16string.h')
-        includes.add('rosidl_generator_c/u16string_functions.h')
+        member_includes.add('rosidl_generator_c/u16string.h')
+        member_includes.add('rosidl_generator_c/u16string_functions.h')
 
     if isinstance(type_, NamespacedType):
         namespaced_types.add(get_jni_type(type_))
-        includes.add(idl_structure_type_to_c_include_prefix(type_) + '.h')
+        include_prefix = idl_structure_type_to_c_include_prefix(type_)
+        # TODO(jacobperron): Remove this logic after https://github.com/ros2/rosidl/pull/432 (Foxy)
+        # Strip off any service or action suffix
+        if include_prefix.endswith('__request'):
```

Comment:
Could you add a comment explaining the rationale for the following code?

Reply:
Added a comment: 97d0ead

IMO, it's an implementation detail from lower in the rosidl pipeline. Hence the TODO where we can remove this logic from Foxy onward.